src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1586,6 +1587,39 @@ else if (null == currentPrimaryPlaceHolder) {
                     currentConnectPlaceHolder = currentPrimaryPlaceHolder;
                 }
 
+                // DNS resolution to retrieve all IP addresses if not done yet
+                if (null == inetAddrs) {
+                    try {
+                        inetAddrs = InetAddress.getAllByName(currentConnectPlaceHolder.getServerName());
+                    }
+                    catch (IOException ex) {
+                        SQLServerException.ConvertConnectExceptionToSQLServerException(currentConnectPlaceHolder.getServerName(),
+                                currentConnectPlaceHolder.getPortNumber(), this, ex);
+                    }
+                }
+
+                // if only one IP is resolved and user didn't set TNIR explicitly then set TNIR to false
+                if ((1 == inetAddrs.length) && !userSetTNIR) {
+                    useTnir = false;
+                }
+
+                //calculate timeout interval for the first attempt
+                if(0 == attemptNumber){
```

Comment:
Nit: Run the changed code through code formatter. There are some spacing differences with the rest of the file.